wodles/aws/aws-s3.py

```diff
@@ -718,22 +864,253 @@ def load_information_from_file(self, log_key):
+
+    def get_days_since_today(self, date):
+        date = datetime.strptime(date, "%Y%m%d")
+        # it is necessary to add one day
+        delta = datetime.utcnow() - date + timedelta(days=1)
+        return delta.days
+
+    def get_date_list(self, aws_account_id, aws_region, flow_log_id):
+        num_days = self.get_days_since_today(self.get_date_last_log(aws_account_id, aws_region, flow_log_id))
+        date_list_time = [datetime.utcnow() - timedelta(days=x) for x in range(0, num_days)]
+        return [datetime.strftime(date, "%Y/%m/%d") for date in reversed(date_list_time)]
+
+    def get_date_last_log(self, aws_account_id, aws_region, flow_log_id):
+        try:
+            query_date_last_log = self.db_connector.execute(self.sql_get_date_last_log_processed.format(
+                                                                                        table_name=self.db_table_name,
+                                                                                        aws_account_id=aws_account_id,
+                                                                                        aws_region=aws_region,
+                                                                                        flow_log_id=flow_log_id))
+            last_date_processed = query_date_last_log.fetchone()[0]
+        # if DB is empty
+        except TypeError as e:
+            last_date_processed = datetime.strftime(datetime.utcnow(), "%Y%m%d") # get today date if DB is empty
+        return str(last_date_processed)
+
+    def iter_regions_and_accounts(self, account_id, regions):
+        if not account_id:
+            # No accounts provided, so find which exist in s3 bucket
+            account_id = self.find_account_ids()
+        for aws_account_id in account_id:
+            # No regions provided, so find which exist for this AWS account
+            if not regions:
+                regions = self.find_regions(aws_account_id)
+                if regions == []:
+                    continue
+            for aws_region in regions:
+                debug("+++ Working on {} - {}".format(aws_account_id, aws_region), 1)
+                # for each flow log id
+                for flow_log_id in self.flow_logs_ids:
+                    date_list = self.get_date_list(aws_account_id, aws_region, flow_log_id)
+                    for date in date_list:
+                        self.iter_files_in_bucket(aws_account_id, aws_region, date, flow_log_id)
+                self.db_maintenance(aws_account_id, aws_region, flow_log_id)
+
+    def db_maintenance(self, aws_account_id, aws_region, flow_log_id):
+        debug("+++ DB Maintenance", 1)
+        try:
+            self.db_connector.execute(self.sql_db_maintenance.format(
+                table_name=self.db_table_name,
+                aws_account_id=aws_account_id,
+                aws_region=aws_region,
+                flow_log_id=flow_log_id,
```

Comment:
This function is almost the same as:
https://github.com/wazuh/wazuh/blob/0c7ada0db1e4b4d49eb99e9f92ea873d8c13dfa6/wodles/aws/aws-s3.py#L406
The only difference is using flow_log_id as field. Isn't there a way to avoid copying and pasting code?

Reply:
DB table for `VPS` has one more column (`flow_log_id`) than the table of `CloudTrail`. I think is better not change this by now.